x/auth/keeper/keeper.go

```diff
@@ -278,3 +281,35 @@ func (ak AccountKeeper) GetParams(ctx context.Context) (params types.Params) {
 	}
 	return params
 }
+
+func (ak AccountKeeper) AsyncMsgsExec(ctx context.Context, signer sdk.AccAddress, msgs []sdk.Msg) ([][]byte, []error) {
+	results := make([][]byte, len(msgs))
+	errors := make([]error, len(msgs))
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+
+	for i, msg := range msgs {
```

Comment:
we must check that the msg sender is equal to signer, in order to avoid the signer impersonating.